### PR TITLE
Improvements to service extensions.

### PR DESCRIPTION
**Pull request by SebastianStehle**

### CODE REVIEW

File: src/NSwag.AspNetCore/Extensions/NSwagServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -28,21 +29,55 @@ public static IServiceCollection AddOpenApiDocument(this IServiceCollection serv
...
-                configure?.Invoke(settings);
+                configure?.Invoke(settings, s);
 
                 if (settings.PostProcess != null)
                 {
                     var processor = new ActionDocumentProcessor(context => settings.PostProcess(context.Document));
                     settings.DocumentProcessors.Add(processor);
                 }
 
+                foreach (var documentProcessor in s.GetRequiredService<IEnumerable<IDocumentProcessor>>())
```
Comment (RSuter) [inline src/NSwag.AspNetCore/Extensions/NSwagServiceCollectionExtensions.cs]: Doesn't this fail if no processors are registered?

File: src/NSwag.AspNetCore/Extensions/NSwagServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -28,21 +29,55 @@ public static IServiceCollection AddOpenApiDocument(this IServiceCollection serv
         /// <summary>Adds services required for Swagger 2.0 generation (change document settings to generate OpenAPI 3.0).</summary>
         /// <param name="serviceCollection">The <see cref="IServiceCollection"/>.</param>
         /// <param name="configure">Configure the document.</param>
-        public static IServiceCollection AddSwaggerDocument(this IServiceCollection serviceCollection, Action<SwaggerDocumentSettings> configure = null)
```
Comment (RSuter) [inline src/NSwag.AspNetCore/Extensions/NSwagServiceCollectionExtensions.cs]: Please add this overloda for backward compat.

File: src/NSwag.AspNetCore/Extensions/NSwagServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -28,21 +29,55 @@ public static IServiceCollection AddOpenApiDocument(this IServiceCollection serv
...
+        /// <param name="configure">Configure the document.</param>
+        public static IServiceCollection AddSwaggerDocument(this IServiceCollection serviceCollection, Action<SwaggerDocumentSettings, IServiceProvider> configure = null)
         {
             serviceCollection.AddSingleton(s =>
             {
                 var settings = new SwaggerDocumentSettings();
                 settings.SchemaType = SchemaType.Swagger2;
 
-                configure?.Invoke(settings);
+                configure?.Invoke(settings, s);
```
Comment (RSuter) [inline src/NSwag.AspNetCore/Extensions/NSwagServiceCollectionExtensions.cs]: Rename s to services

**Comment by RSuter** [comment]
Thanks for the pr. Ive added some comments

File: src/NSwag.AspNetCore/Extensions/NSwagServiceCollectionExtensions.cs (hunk 1 above)
Comment (SebastianStehle) [reply src/NSwag.AspNetCore/Extensions/NSwagServiceCollectionExtensions.cs]: No, it does not.

**Comment by RSuter** [comment]
v12.0.4

**Pull request merged**